### PR TITLE
Change NDArray ndims to ndim

### PR DESCRIPTION
**Pull request by johnkerl**
Addresses #84

### CODE REVIEW

File: python-spec/src/somacore/data.py [hunk 1]
```diff
@@ -88,8 +88,9 @@ def shape(self) -> Tuple[int, ...]:
         raise NotImplementedError()
 
     @property
-    def ndims(self) -> int:
-        """The Number of Dimensions in this array."""
+    @abc.abstractmethod
```
Comment (thetorpedodog) [inline python-spec/src/somacore/data.py:91]: Since this is the implementation of ndim, this should not be marked as abstract.

**Review by thetorpedodog — APPROVED** [review]

**Review by bkmartinjr — APPROVED** [review]
LGTM. 

May also want to update `abstract_specfication.md`

File: python-spec/src/somacore/data.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def shape(self) -> Tuple[int, ...]:
     @abc.abstractmethod
     def ndim(self) -> int:
         """The number of dimensions in this array."""
-        return len(self.shape)
+        raise NotImplementedError()
```
Comment (thetorpedodog) [inline python-spec/src/somacore/data.py:94]: This should still be `len(self.shape)`; there’s no need to make downstream users implement this.

**Review by thetorpedodog — CHANGES_REQUESTED** [review]

File: python-spec/src/somacore/data.py (hunk 2 above)
Comment (thetorpedodog) [reply python-spec/src/somacore/data.py:94]: Did not see the earlier comment.

Comment (thetorpedodog) [reply python-spec/src/somacore/data.py]: Actually it looks like this still can be `len(self.shape)`. An implementation *can* override this, but it should always be valid.

**Review by thetorpedodog — APPROVED** [review]

File: python-spec/src/somacore/data.py (hunk 2 above)
Comment (thetorpedodog) [reply python-spec/src/somacore/data.py]: What would need changing there? It returns a tuple whose length will be `ndim`, so taking its `len` should be fine (as the spec implies). Am I missing something?

**Pull request merged**